CanvasColorSpaceProposal.md

```diff
@@ -103,61 +84,48 @@ canvas.getContext('2d', { colorSpace: "rec2020",
+* All blending operations are performed in the canvas' backing storage color space.
+  * The midpoint between the backbuffer's colors ``(1,0,0)`` and ``(0,1,0)`` will be ``(0.5,0.5,0)``, even though this is neither the physical nor perceptual midpoint between the two colors.
+  * This means that the same content, rendered with two different canvas backing stores, may be slightly different.
+* The functions ``toDataURL`` and ``toBlob`` should produce resources that best match the fidelity of the underlying canvas.
+  * Subject to the limitations of the implementation and the requested format.
 
-#### Selecting the best encoding for the user agent's display device
-Selection should be based on the best color space match (see above). For srgb, at least 8 bits per component is recommended; for p3, 10 bits; and for rec2020, 12 bits.  The float16 format is suitable for any colorspace.  There may soon be a proposal to add a way of detecting HDR displays, possibly something like "window.screen.isHDR()" (TBD), which would be a good hint to use the float16 format.
+#### WebGL behavior
 
-#### Non-standard color spaces
-For future consideration: support could be added for color space defined using the [CSS @color-profile rule](https://www.w3.org/TR/css-color-4/#at-profile).
+* Values stored in WebGL backbuffers are in the canvas's color space.
+* Values written by ``gl_FragColor`` use the primaries of the canvas' color space.
+* For the color encoding of ``"unorm-srgb8"``, the interpretation of values written by ``gl_FragColor`` is as follows:
+  * The color value is in a linear color space.
+  * The inverse sRGB transfer function is applied only after blending has occurred.
+  * E.g, if the value ``0.5`` is written in ``gl_FragColor``, the result will be stored as ``0xbc``.
+* For all other color encodings, the value written by ``gl_FragColor`` is in the canvas' color space.
 
 #### Compositing the canvas element
-Canvas contents are composited in accordance with the canvas element's style (e.g. CSS compositing and blending rules). The necessary compositing operations must be performed in an intermediate colorspace, the compositing space, that is implementation specific. The compositing space must have sufficient precision and a sufficiently wide gamut to guarantee no undue loss of precision or gamut clipping in bringing the canvas's contents to the display. Implementations should not expose color spaces that are unreasonble for the display.
+
+Canvas contents are composited in accordance with the canvas element's style (e.g. CSS compositing and blending rules). The necessary compositing operations must be performed in an intermediate colorspace, the compositing space, that is implementation specific. The compositing space must have sufficient precision and a sufficiently wide gamut to guarantee no undue loss of precision or gamut clipping in bringing the canvas's contents to the display.
```

Comment:
This is great I think, though removing "Implementations should not expose color spaces that are unreasonble for the display" is a big change!

Reply:
Removing "Implementations should not expose color spaces that are unreasonble for the display" is an example of the old cruft in the spec that should be removed (the original authors are long gone, and the context in which they wrote this long forgotten).

My argument for this is: A user should be able to author and manipulate content that current monitor happens to support. An example is a web app being used to crop/rotate photos using 2D canvas. If a P3 image is uploaded, then it should be manipulatable without crushing it to sRGB, regardless of the user's monitor capabilities.

Reply:
I think I support this change, but I would prefer to see it in a separate non-cleanup PR, if you can!

Reply:
@RafaelCintron is interested in how/if we handle monitor-non-native spaces, I think.

Reply:
Left it here for now, and I'll follow up to remove that sentence and add motivation.